parsl/executors/high_throughput/executor.py

```diff
@@ -713,8 +713,20 @@ class BlockInfo:
             tasks: int  # sum of tasks in this block
             idle: float  # shortest idle time of any manager in this block
 
+        # block_info will be populated from two sources:
+        # the Job Status Poller mutable block list, and the list of blocks
+        # which have connected to the interchange.
+
+        def new_block_info():
+            return BlockInfo(tasks=0, idle=float('inf'))
+
+        block_info: Dict[str, BlockInfo] = defaultdict(new_block_info)
+
+        for block_id, job_status in self._status.items():
+            if job_status.state not in TERMINAL_STATES:
+                block_info[block_id] = new_block_info()
```

Comment:
Per the PR comment to me, I think this looks fine.  The only suggestion I might make is to take advantage of the `dataclass`'s default handling via:
```python
@dataclass
class BlockInfo:
    tasks: int = 0
    idle: float = float('inf')
```
And then there's no need for the `new_block_info` function.  Just:
```python
block_info = defaultdict(BlockInfo)
```
Beyond that, per our side-band conversation, I don't think now's the time to tackle the code smell you've discovered.  This whole method could be cleaner yes, but suggest "not now."